packages/cli/src/zosjobs/submit/Submit.shared.handler.ts

```diff
@@ -31,150 +31,167 @@ export default class SharedSubmitHandler extends ZosmfBaseHandler {
+
+                        if (parms.viewAllSpoolContent) {
+                            spoolFilesResponse = apiObj;
+                        }
                     }
 
                     break;
-                } else {
-                    response = await SubmitJobs.submitJobCommon(this.mSession, {jobDataSet: this.mArguments.dataset,
+                // Submit JCL from a USS file
+                case "uss-file":
+                    response = await SubmitJobs.submitJobCommon(this.mSession, {jobUSSFile: this.mArguments.file,
                         jclSymbols: this.mArguments.jclSymbols});
                     apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
-                    source = this.mArguments.dataset;
+                    source = this.mArguments.ussfile;
 
                     if (parms.viewAllSpoolContent) {
                         spoolFilesResponse = apiObj;
                     }
-                }
-
-                break;
-            // Submit JCL from a USS file
-            case "uss-file":
-                response = await SubmitJobs.submitJobCommon(this.mSession, {jobUSSFile: this.mArguments.file,
-                    jclSymbols: this.mArguments.jclSymbols});
-                apiObj = await SubmitJobs.checkSubmitOptions(this.mSession, parms, response);
-                source = this.mArguments.ussfile;
-
-                if (parms.viewAllSpoolContent) {
-                    spoolFilesResponse = apiObj;
-                }
 
-                break;
-            // Submit the JCL from a local file
-            case "local-file": {
-                parms.jclSource = this.mArguments.localFile;
-                const JclString = fs.readFileSync(this.mArguments.localFile).toString();
-                apiObj = await SubmitJobs.submitJclString(this.mSession, JclString, parms);
-                source = this.mArguments.localFile;
-                if (parms.viewAllSpoolContent) {
-                    spoolFilesResponse = apiObj;
+                    break;
+                // Submit the JCL from a local file
+                case "local-file": {
+                    parms.jclSource = this.mArguments.localFile;
+                    let JclString: string;
+                    try {
+                        JclString = fs.readFileSync(this.mArguments.localFile).toString();
+                    } catch (err) {
+                        throw new ImperativeError({
+                            msg: ZosJobsMessages.nodeJsFsError.message,
+                            additionalDetails: err.toString(),
+                            causeErrors: err
+                        });
+                    }
+                    if (parms.viewAllSpoolContent) {
+                        spoolFilesResponse = apiObj;
+                    }
+                    apiObj = await SubmitJobs.submitJclString(this.mSession, JclString, parms);
+                    source = this.mArguments.localFile;
```

Comment:
The `apiObj` definition needs to be moved above the `if` statement so that we don't assign an undefined value to `spoolFilesResponse`

Reply:
After making this change, you should be able to uncomment line 615 in the unit test and have it pass 🙂